contributors/design-proposals/storage/csi-migration.md

```diff
@@ -287,8 +287,38 @@ existing Pods in the ADC.
 
 
 ### Volume Resize
-
-TODO: Design
+#### Offline Resizing
+In the OperationGenerator, `GenerateExpandVolumeFunc` is used to expand a volume after
+user updates a PVC requesting more storage. At the beginning of the `GenerateExpandVolumeFunc`, we will check whether
+the volume supports migration and whether migration is enabled. If enabled, we will return no-op function.
+
+In the in-tree resize controller, we will create a new PVC annotation `volume.kubernetes.io/storage-resizer` and set the value
+to the name of resizer. If the PV is CSI PV or migrated in-tree PV, the annotation will be set to the name of CSI driver; 
+otherwise, it will be set to the name of in-tree plugin.
+
+For migrated volume, The CSI resizer name will be derived from translating in-tree plugin name
+to CSI driver name by translation library. We will also add an event to PVC about resizing being handled
+by external controller.
+
+In the OperationGenerator, `GenerateMountVolumeFunc` is used to expand file system after volume
```

Comment:
Maybe separate this section into Controller Expansion, and Filesystem Expansion.  This paragraph is slightly confusing because it's sitting in the middle of the controller expansion discussion.

Reply:
updated